src/cbmc/bmc.cpp

```diff
@@ -596,3 +599,93 @@ void bmct::setup_unwind()
   if(options.get_option("unwind")!="")
     symex.set_unwind_limit(options.get_unsigned_int_option("unwind"));
 }
+
+int bmct::do_language_agnostic_bmc(
+    const optionst &opts,
+    const goto_modelt &goto_model,
+    const ui_message_handlert::uit &ui,
+    messaget *const message)
+{
+  cbmc_solverst solvers(
+      opts,
+      goto_model.symbol_table,
+      message->get_message_handler());
+  solvers.set_ui(ui);
+
+  message_handlert &mh=message->get_message_handler();
+  safety_checkert::resultt result;
+  goto_symext::branch_worklistt worklist;
+  try
+  {
+    {
+      std::unique_ptr<cbmc_solverst::solvert> cbmc_solver;
+      cbmc_solver=solvers.get_solver();
+      prop_convt &pc=cbmc_solver->prop_conv();
+      bmct bmc(opts, goto_model.symbol_table, mh, pc, worklist);
+      bmc.set_ui(ui);
+      result=bmc.run(goto_model.goto_functions);
+    }
+    INVARIANT(opts.get_bool_option("paths") || worklist.empty(),
+        "Some branches were unexplored when model-checking "
```

Comment:
Recommend making a "should" statement here, as otherwise it's often confusing whether the statement says what should have been or what went wrong.